### PR TITLE
fix: template not supporting to use rest-spread

### PR DESCRIPTION
**Pull request by vip30**
fix: https://github.com/JuniorTour/vue-template-babel-compiler/issues/9

### CODE REVIEW

File: src/renderCompiler.js [hunk 1]
```diff
@@ -21,7 +21,8 @@ export function renderCompiler(code, options) {
       '@babel/plugin-proposal-optional-chaining',
       '@babel/plugin-transform-block-scoping',
       '@babel/plugin-transform-destructuring',
-      '@babel/plugin-transform-spread',
+      "@babel/plugin-transform-spread",
+      ["@babel/plugin-proposal-object-rest-spread", { loose: true, useBuiltIns: true }],
```
Comment (JuniorTour) [inline src/renderCompiler.js:25]: Nice config for combining `loose && useBuiltIns`!

File: lib/index.js [hunk 2]
```diff
@@ -104,7 +104,10 @@ function renderCompiler(code, options) {
     filename: 'compiledTemplate',
     // not enable strict mode, in order to parse WithStatement
     sourceType: 'script',
-    plugins: ['@babel/plugin-proposal-nullish-coalescing-operator', '@babel/plugin-proposal-optional-chaining', '@babel/plugin-transform-block-scoping', '@babel/plugin-transform-destructuring', '@babel/plugin-transform-spread', '@babel/plugin-transform-arrow-functions', '@babel/plugin-transform-parameters', parseWithStatementToVm]
+    plugins: ['@babel/plugin-proposal-nullish-coalescing-operator', '@babel/plugin-proposal-optional-chaining', '@babel/plugin-transform-block-scoping', '@babel/plugin-transform-destructuring', "@babel/plugin-transform-spread", ["@babel/plugin-proposal-object-rest-spread", {
```
Comment (JuniorTour) [inline lib/index.js:107]: We need add `@babel/plugin-proposal-object-rest-spread` to package.json `dependencies`.

Use `yarn add @babel/plugin-proposal-object-rest-spread` 

And commit `yarn.lock` together.

Comment (vip30) [reply lib/index.js:107]: https://babeljs.io/docs/en/babel-plugin-proposal-object-rest-spread
base on the docs, 

It should be included in the preset-env
```
NOTE: This plugin is included in @babel/preset-env, in ES2018
```

Comment (JuniorTour) [reply lib/index.js:107]: Yes, you are right, `@babel/plugin-proposal-object-rest-spread` is a part of `@babel/preset-env`.

But this repo don't depend on `@babel/preset-env`, it is just a `devDependencies`.

When we only install this lib, `@babel/preset-env` will not get installed, 
微信截图_20210914013426

If we don't add new plugin to `dependencies`，user may get Error:
``` js
Error: [BABEL]: Cannot find module '@babel/core'
Require stack:
- ..\node_modules\@babel\plugin-proposal-object-rest-spread\lib\index.js
//...
```

Comment (vip30) [reply lib/index.js:107]: You're correct
Already added back the dependency

**Pull request merged**